src/main/java/br/edu/ufcg/computacao/alumni/core/Match.java

```diff
@@ -0,0 +1,123 @@
+package br.edu.ufcg.computacao.alumni.core;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import br.edu.ufcg.computacao.alumni.api.http.response.LinkedinAlumnusData;
+import br.edu.ufcg.computacao.alumni.api.http.response.UfcgAlumnusData;
+import br.edu.ufcg.computacao.alumni.core.holders.LinkedinDataHolder;
+import br.edu.ufcg.computacao.alumni.core.models.Curso;
+import br.edu.ufcg.computacao.alumni.core.models.Grau;
+import br.edu.ufcg.computacao.alumni.core.models.LinkedinSchoolData;
+
+public class Match {
+
+	public Match() {
+	}
+
+	private Curso getCurso(String field, String degree) {
+		field = field.toUpperCase().trim();
+		degree = degree.toUpperCase().trim();
+
+		if ("".equals(field) || "".equals(degree)) {
+			return null;
+		}
+		if (field.contains("COMPUTER") && (degree.contains("MASTER") || degree.contains("MSC"))) {
+			return Curso.MESTRADO_EM_CIENCIA_DA_COMPUTACAO;
+		}
+		if (field.contains("COMPUTER")
+				&& (degree.contains("BC") || degree.contains("BACHELOR") || degree.contains("BS"))) {
+			return Curso.GRADUACAO_CIENCIA_DA_COMPUTACAO;
+		}
+		if (field.contains("COMPUT") && degree.contains("PH")) {
+			return Curso.DOUTORADO_EM_CIENCIA_DA_COMPUTACAO;
+		}
+		if (field.contains("INFORMAT") && (degree.contains("MASTER") || degree.contains("MSC"))) {
+			return Curso.MESTRADO_EM_INFOMATICA;
+		}
+		if (field.contains("") && (degree.contains("BC") || degree.contains("BACHELOR") || degree.contains("BS"))) {
+			return Curso.GRADUACAO_PROCESSAMENTO_DE_DADOS;
+		}
+		return null;
+	}
+
+	private int getScoreFromName(String alumniName, String linkedinName) {
+		if (alumniName.equals(linkedinName))
+			return 1;
+		return 0;
+	}
+
+	private int getScoreFromCurso(List<Curso> alumniCursos, Curso linkedinCurso) {
+		if (linkedinCurso == null)
+			return 0;
+		if (alumniCursos.contains(linkedinCurso))
+			return 1;
+		return 0;
+	}
+
+	private int getScoreFromSchoolUrl(String schoolUrl, String linkedinUrl) {
+		if (linkedinUrl == null)
+			return 0;
+		if (schoolUrl.equals(linkedinUrl))
```

Comment:
Esse também é um match mais complicado por conta da criação da UFCG. Ou seja, temos egressos que foram formados pela UFPB (eu, por exemplo). Se a formatura for até 2002, precisamos considerar a UFPB também. Mas, por enquanto, fica assim.